src/main/kotlin/org/jitsi/nlj/transform/node/incoming/TccGeneratorNode.kt

```diff
@@ -89,6 +91,17 @@ class TccGeneratorNode(
      */
     private fun addPacket(tccSeqNum: Int, timestamp: Long, isMarked: Boolean, ssrc: Long) {
         synchronized(lock) {
+            if (packetArrivalTimes.ceilingKey(windowStartSeq) == null) {
+                // Start new feedback packet, cull old packets.
+                packetArrivalTimes.entries.removeIf {
+                    entry -> entry.key < tccSeqNum &&
+                        timestamp - entry.value >= BACK_WINDOW_MS
```

Comment:
This is a bit concerning because:
1. The map will usually contain ~500ms worth of packets (~150-200 packets).
2. `removeIf` is linear
3. `removeIf` will execute once per sent TCC feedback

Are any of my premises wrong?

Reply:
I don't think they are. I'll try profiling it and see what I find.

Chrome does the same thing (it uses an explicit for loop with an iterator rather than a lambda but the semantic is the same) but of course it's only sending one source, so presumably less media, and also the computational cost compared to a browser is relatively much less than an SFU.

In theory if we used an explicit loop we could terminate the search by `entry.key`, but I'm pretty sure that in the normal cases this wouldn't be much of a win since tccSeqNum will be greater than any previous key anyway.